<!--
  Nodes common to all simulators:
  - cmd_vel multiplexer
  - virtual sensors
  - navigation
 -->

<launch>
  <arg name="move_base_flex"  default="true"/>
  <arg name="base_frame_id"   default="base_footprint"/>
  <arg name="odom_frame_id"   default="odom"/>
  <arg name="global_frame_id" default="map"/>
  <arg name="smooth_velocity" default="false"/>  <!-- hampers navigation in simulation -->
  <arg name="save_robot_pose" default="false"/>  <!-- senseless in simulation -->
  <arg name="localization"    default="amcl"/>   <!-- amcl, static, gazebo -->

  <!-- Name of the map to use (without path nor extension) and initial position -->
  <arg name="map_name"        default="home"/>
  <arg name="map_file"        default="$(find thorp_navigation)/resources/maps/$(arg map_name).yaml"/>
  <arg name="initial_pose_x"  default="0.0"/>
  <arg name="initial_pose_y"  default="0.0"/>
  <arg name="initial_pose_a"  default="0.0"/>

  <!--  ******** Velocity commands multiplexer ********  -->
  <node pkg="nodelet" type="nodelet" name="mobile_base_nodelet_manager" args="manager"/>
  <node pkg="nodelet" type="nodelet" name="cmd_vel_mux" args="load yocs_cmd_vel_mux/CmdVelMuxNodelet mobile_base_nodelet_manager">
    <param name="yaml_cfg_file" value="$(find thorp_bringup)/param/vel_multiplexer.yaml"/>
  </node>

  <!--  ***************** Navigation ******************  -->
  <include file="$(find thorp_navigation)/launch/navigation.launch">
    <arg name="move_base_flex"  value="$(arg move_base_flex)"/>
    <arg name="base_frame_id"   value="$(arg base_frame_id)"/>
    <arg name="odom_frame_id"   value="$(arg odom_frame_id)"/>
    <arg name="global_frame_id" value="$(arg global_frame_id)"/>
    <arg name="smooth_velocity" value="$(arg smooth_velocity)"/>
    <arg name="save_robot_pose" value="$(arg save_robot_pose)"/>
    <arg name="localization"    value="$(arg localization)"/>
    <arg name="map_file"        value="$(arg map_file)"/>
    <arg name="initial_pose_x"  value="$(arg initial_pose_x)"/>
    <arg name="initial_pose_y"  value="$(arg initial_pose_y)"/>
    <arg name="initial_pose_a"  value="$(arg initial_pose_a)"/>
  </include>
</launch>
